resources/domain/request/index.ts

```diff
@@ -1,3 +1,31 @@
+import { SignTypedDataVersion } from '@metamask/eth-sig-util'
+import type { SignatureRequestType, TypedMessage } from '../../../main/accounts/types'
+
 export const isCancelableRequest = (status: string): Boolean => {
   return !['sent', 'sending', 'verifying', 'confirming', 'confirmed', 'error', 'declined'].includes(status)
 }
+
+const eip2612PermitTypes = [
```

Comment:
we can probably break this out into its own module otherwise this one if going to become quite bloated